pkg/canary/comparator/comparator.go

```diff
@@ -186,12 +244,116 @@ func (c *Comparator) run() {
 			c.entrySent(e)
 		case <-t.C:
 			c.pruneEntries()
+		case <-sc.C:
+			// Only run one instance of spot check at a time.
+			c.spotMtx.Lock()
+			if !c.spotCheckRunning {
+				c.spotCheckRunning = true
+				go c.spotCheckEntries(time.Now())
+			}
+			c.spotMtx.Unlock()
+		case <-mt.C:
+			// Only run one intstance of metric tests at a time.
+			c.metTestMtx.Lock()
+			if !c.metricTestRunning {
+				c.metricTestRunning = true
+				go c.metricTest(time.Now())
+			}
+			c.metTestMtx.Unlock()
 		case <-c.quit:
 			return
 		}
 	}
 }
 
+func (c *Comparator) metricTest(currTime time.Time) {
+	// Always make sure to set the running state back to false
```

Comment:
wouldn't it be better to take the lock here instead of outside ?

Reply:
the reason I didn't do it that way is because it would start the goroutine and realize it's already running and exit the goroutine, so this approach saves spawning a goroutine which would just exit

Reply:
i could add another level of function redirection to put the lock taking in the metricTest code which then does the locking and checking before launching the internals in an anonymous function or another function, maybe makes it easier to read?